tests/js/searchtools.js

```diff
@@ -147,6 +147,20 @@ describe('Basic html theme search', function() {
       checkRanking(expectedRanking, results);
     });
 
+    it('should score a main-title match above an index entry match', function() {
```

Comment:
I don't think that this test case accurately reflects the intent of the functionality desscribed in #11696; `main` index entries are [denoted by an exclamation mark](https://github.com/sphinx-doc/sphinx/blob/174348677617c31008cbd504c608cb2c5fd66f36/sphinx/util/nodes.py#L375-L377), and the test data here doesn't reflect how those should rank compared to other items in `indexentries`.  I'll rework the test and fixture data.

Reply:
> I'll rework the test and fixture data.

No; let's just simplify and isolate the optimizations from the test updates.  The test updates can be applied separately.